tt_metal/impl/buffers/buffer.cpp

```diff
@@ -131,7 +131,7 @@ Buffer::Buffer(
     bottom_up_(bottom_up),
     buffer_page_mapping_(nullptr),
     allocate_(allocate) {
-    TT_FATAL(this->device_ != nullptr and this->device_->allocator_ != nullptr);
+    TT_FATAL(this->device_ != nullptr and this->device_->allocator_ != nullptr, "Device and allocator need to not be null.");
```

Comment:
Would be best to have 2 separate checks and separate messages one for device one for allocator

Reply:
I left the conditions as is. I prefer to have refactor PRs not change the logic. If a code owner sees this and thinks it's worthwhile fixing, they can do that separately.